src/CMakeLists.txt

```diff
@@ -56,18 +56,41 @@ find_package(Threads REQUIRED)
 set(Boost_USE_MULTITHREADED ON)
 
 add_library(_espressopp ${ESPRESSO_SOURCES})
+
+# compiler specific configurations
+if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
+    if(ESPP_LOCAL_ARCHITECTURE)
+        target_compile_options(_espressopp PRIVATE "-march=native")
+    endif()
+endif()
+if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
+    if(ESPP_LOCAL_ARCHITECTURE)
+        target_compile_options(_espressopp PRIVATE "-march=native")
+    endif()
+endif()
+if(CMAKE_CXX_COMPILER_ID MATCHES Intel)
+    if(ESPP_LOCAL_ARCHITECTURE)
+        target_compile_options(_espressopp PRIVATE "-xHost")
```

Comment:
Can you also add `-restrict` here?

Reply:
Interesting. Is my understanding correct that Intel uses this flag to enable checking for the `restrict` keyword? I hope it does not transform all pointers into `restrict`. And it only applies to `restrict`, `__restrict` is always enabled? Pretty weird,
https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/compiler-options/compiler-option-details/language-options/restrict-qrestrict.html

Reply:
added

Reply:
I think it's the latter. I use a lot of them in the pointer declarations in the vec submodule.

Reply:
My state is, I do think we do not need it but it does not hurt. So we can enable it.

Reply:
There is even `-qopt-zmm-usage=high` recommended for AVX-512 which slightly improves loop vectorization. But we can also decide to add it later.